File: sdks/php/rest/src/Types/WebhookListResponse.php
<?php

namespace Brume\Rest\Types;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;
use Brume\Rest\Core\Types\ArrayType;

/**
 * `GET /v1/webhooks` success response.
 */
class WebhookListResponse extends JsonSerializableType
{
    /**
     * @var array<WebhookSummary> $webhooks
     */
    #[JsonProperty('webhooks'), ArrayType([WebhookSummary::class])]
    public array $webhooks;

    /**
     * @param array{
     *   webhooks: array<WebhookSummary>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->webhooks = $values['webhooks'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: sdks/csharp/rest/src/Brume.Rest/Types/WebhookListResponse.cs
using Brume.Rest.Core;
using global::System.Text.Json;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

/// <summary>
/// `GET /v1/webhooks` success response.
/// </summary>
[Serializable]
public record WebhookListResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("webhooks")]
    public IEnumerable<WebhookSummary> Webhooks { get; set; } = new List<WebhookSummary>();

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
